plugins/language_fsharp.lua

```diff
@@ -0,0 +1,108 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "F#",
+  files = { "%.fsi?$" },
+  comment = "//",
+  patterns = {
+    { pattern = "//.*",                                type = "comment"   }, -- Single-line comment
+    { pattern = { '"', '"', '\\' },                    type = "string"    }, -- String, quotation marks
+    { regex   = "\\${1,2}(?=\"?\'?)",                  type = "string"    }, -- String, $
+    { pattern = "-?0x%x+",                             type = "number"    }, -- ?
+    { pattern = "-?%d+[%deE]*f?",                      type = "number"    }, -- ?
+    { pattern = "-?%.?%d+f?",                          type = "number"    }, -- ?
+    { regex   = "\\<.+\\>",                            type = "keyword2"  }, -- Generic type
+    { regex   = "\\[\\<.+\\>\\]",                      type = "keyword2"  }, -- Attribute
+    { regex   = "\\:\\s?\\w+",                         type = "keyword2"  }, -- Type
+    { regex   = "\\_(?=\\s?\\:)",                      type = "normal"    }, -- _ should be normal when used as ?
+    { pattern = "[%+%-=/%*%^%%<>!~|&_:]",              type = "operator"  }, -- Operators
+    { regex   = [[\.{2}\<?\s?(?=[\\-]?[a-z0-9])]],     type = "operator"  }, -- Range operators
+    { regex   = [[\-\>(?=\s)]],                        type = "function"  }, -- Function arrow
+    { regex   = "\\w+(?=\\s?[(])",                     type = "function"  }, -- Function without generic type
+    { regex   = "\\w+()\\s?\\[?\\<\\'?\\w+\\>\\]?",    type = { "function", "keyword2" } }, -- Function with generic type
+    { regex   = "\\#\\w+",                             type = "keyword2"  }, -- Load
+    { regex   = "\\'\\w+",                             type = "keyword"   }, -- Special variable
+  },
```

Comment:
Should this have the basic catch-all-words pattern?

Reply:
Yes, actually. That pattern is for variables that begin with `'`.

Reply:
I meant that this syntax should have a catch-all at the end, not that regex specifically.

Reply:
I don't think it need one.

Reply:
What is matching with the symbols in the symbols table?

Reply:
Ah wait, it needs one. Because the symbols are words.

Reply:
Thanks.

Reply:
No worries.